driver/SDL/SDL_keyboard.c

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file keyboard.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "SDL_keyboard.h"
+#if USE_KEYBOARD != 0
+
+/*********************
+ *      DEFINES
+ *********************/
+#ifndef MONITOR_ZOOM
+#define MONITOR_ZOOM    1
+#endif
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+static bool keyup = false;
+static bool keydown = false;
+static bool repeat = false;
+static bool keyboard_was_read = true;
+static int last_key = -1;
+
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+/**
+ * Get the current key and state of the keyboard
+ * @param data store the keyboard data here
+ * @return false
+ */
+void keyboard_read(struct _lv_indev_drv_t * indev_drv, lv_indev_data_t * data)
+{
+    if(!keyboard_was_read)
+    {
+        /* Store the collected data */
+        if(keyup)
+        {
+            data->state = LV_INDEV_STATE_PR;
+            data->key = last_key;
+        }
+        else if(keydown)
+        {
+            data->state = LV_INDEV_STATE_REL;
+            data->key = last_key;
+        }
+    }
+    if(keydown && repeat){
+        data->state = LV_INDEV_STATE_PR;
+        data->key = last_key;
```

Comment:
If you update `data->key = last_key` the same way in all cases, you can move it outside the "if" block.